src/System.Numerics.Tensors/System/Numerics/Tensor.cs

```diff
@@ -956,9 +947,29 @@ private IEnumerable<T> Enumerate()
 
         void ICollection.CopyTo(Array array, int index)
         {
-            for(int i = 0; i < length; i++)
+            if (array == null)
+            {
+                throw new ArgumentNullException(nameof(array));
+            }
+            if (array.Rank != 1)
+            {
+                throw new ArgumentException("Only single dimensional arrays are supported for the requested action.", nameof(array));
+            }
+            if (array.Length < index + Length)
+            {
+                throw new ArgumentException("The number of elements in the Tensor is greater than the available space from index to the end of the destination array.", nameof(array));
+            }
+
+            if (array is T[] destinationArray)
```

Comment:
You already have not null and array size checks in the protected `CopyTo` method. Skip these checks if array is `T[]`.

Reply:
Done.